### PR TITLE
improved limiting code

### LINKED ISSUE #65: Harden matrix/df repr
we have many problems regarding this stuff. i think after #64 has made it in, we should rework this like so:

1. decompose into sliced parts of the df/matrix if larger than limits. e.g.:

	```r
	parts <- list(left = ..., right = ...)
	type <- 'horizontal'
	# or
	parts <- list(full = obj)
	type <- 'full')
	```

2. transform parts into `list` of `matrix` of `character`, similar to:

	```r
	formatted_parts <- lapply(parts, function(part)
		if (is.matrix(part))
			format(part)
		else
			vapply(part, format, character(nrow(part))))
	```

3. glue together parts with ellipses:

	```r
	rv <- switch(type,
		full = parts$full,
		horizontal = cbind(parts$left, ellip.h, parts$right),
		...)
	```

### PR DESCRIPTION
**Pull request by flying-sheep**
fixes #65

### CODE REVIEW

**Comment by flying-sheep** [comment]
hmm, stupid dplyr explicitly sets `row.names` to a sequence starting from 1 if they aren’t set yet.

we need to explicitly set them if `many_rows` is `TRUE`

File: R/repr_matrix_df.r [hunk 1]
```diff
@@ -47,78 +47,79 @@ ellip.limit.vec <- function(v, num, ellip) {
...
+	limitation <- if (many_rows && many_cols) {
+		parts <- list(
+			ul = a[upper, left], ur = a[upper, right],
+			ll = a[lower, left], lr = a[lower, right])
+		'both'
+	} else if (many_rows) {
+		parts <- list(
+			upper = a[upper, , drop = FALSE],
+			lower = a[lower, , drop = FALSE])
+		'vert'
```
Comment (janschulz) [inline R/repr_matrix_df.r]: can we keep the "many_rows" naming? I know `vert` will align the source code, but on the other hand I can see two interpretations for "vertical": "slice it vertically" or "too many stuff in the vertical dimension"... `too_many_rows` is without ambiguity.

Comment (flying-sheep) [inline R/repr_matrix_df.r]: good point!

**Comment by flying-sheep** [comment]
for the record: the checks failed because i deleted the bach so fast

**Pull request merged**